docs/developer/timeout_lock.rst

```diff
@@ -0,0 +1,52 @@
+.. highlight:: python
+
+.. _timeout_lock:
+
+TimeoutLock
+============
+
+Overview
+---------
+
+In OCS, operations are mainly asynchronous, however most agents have
+restrictions on which operations are allowed to be run simultaneously.
+For instance, when only one operation can query with a hardware device at a time
```

Comment:
I think this could be reworded, how about:
"For instance, when an Agent interfaces with a hardware device over serial communication, two commands cannot be issued simultaneously."